math/count_of_trailing_ciphers_in_factorial_n.cpp

```diff
@@ -0,0 +1,64 @@
+/**
+ * @file
+ * @brief A simple program to count the number of ciphers in a factorial of some
+ * number.
+ * @author [Swastika Gupta](https://github.com/swastyy)
+ */
+#include <cassert>
+#include <iostream>
+
+/**
+ * Function to count the number of ciphers.
+ * @param n is the number to be given as argument.
+ * @return it returns the power of 5 in factorial of a given number.
+ */
+
+int count_of_number_of_ciphers_in_factorial_n(uint64_t n) {
+    // count is to store the number of 5's in factorial(n)
+    int count = 0;
+
+    // Keep dividing n by powers of
+    // 5 and update count
+    for (int i = 5; n / i >= 1; i *= 5) {
+        count += static_cast<int>(n) / i;
+    }
+
+    return count;
+}
+
+/** Test function
+ * @returns void
+ */
+void tests() {
+    std::cout << "Test 1:\t n=395\n";
+    if (count_of_number_of_ciphers_in_factorial_n(395) == 97) {
+        std::cout << "passed\n";
+    }
+    std::cout << "failed\n";
+
+    std::cout << "Test 2:\t n=977\n";
+    if (count_of_number_of_ciphers_in_factorial_n(977) == 242) {
+        std::cout << "passed\n";
+    }
+    std::cout << "failed\n";
+
+    std::cout << "Test 3:\t n=871\n";
+    if (count_of_number_of_ciphers_in_factorial_n(871) == 215) {
+        std::cout << "passed\n";
+    }
+    std::cout << "failed\n";
+
+    std::cout << "Test 4:\t n=239\n";
+    if (count_of_number_of_ciphers_in_factorial_n(239) == 57) {
+        std::cout << "passed\n";
+    }
+    std::cout << "failed\n";
```

Comment:
whether the tests passes or not, in both cases `Failed` will be displayed, as it is priting unconditionally.

Reply:
It shoudl be like
```C++
void test() {
    cout << "Running predefined tests...\n";
    assert(count_of_number_of_ciphers_in_factorial_n(871) == 215);
    assert(count_of_number_of_ciphers_in_factorial_n(977) == 242);
...............
............... 
..............// write some more tests covering corner cases
................
...............
    std::cout << "Test Passed...!\n";
}
```

Reply:
Initially I wrote the "failed" thing inside the else block but it was showing warnings then I searched over internet on how to remove these warnings. I saw this article https://clang.llvm.org/extra/clang-tidy/checks/readability-else-after-return.html where its removing else to remove the warnings.

Reply:
Ohh Ohkk! Got it!

Reply:
observe the pattern of execution of what you have written. See if the first test cases passes then program control will go inside the first `if` block and there it prints `passed` and after printing this, program control will come out of the first `if` block and it again gets a print command which will print `failed`, Meaning for a single test it will print both `passed` as well  as `failed` which does not makes any sense. Hope you got it.

Reply:
Yep, thanks for pointing out.

Reply:
@ayaankhan98 I have done the changes. Pl have a look